### ISSUE TITLE
The `block-opening-brace-newline-after` fix has no run of its own over a block holding nothing but comments

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#676]
### What minimal example or steps are needed to reproduce the bug?

`block-opening-brace-newline-after` reads a block holding nothing but comments since [#672](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/672), and reports it, but writes nothing: the run whose first character the check reads is the whitespace in front of the closing brace, the block's own trailing raw, and that run is the one the `block-closing-brace-newline-before`, `block-closing-brace-space-before` and `block-closing-brace-empty-line-before` rules write.

```css
a {/*c*/}
```

`1:4` under `always` — `Expected newline after "{"` — and `--fix` hands the file back as it went in. A fix that writes anywhere else does not clear the warning, since the check reads the run in front of the brace and nothing else: over `a {/*1*/ ⏎ /*2*/}` a break written in front of the first comment never reaches that brace, because the second comment carries a break of its own and the carry stops there.

Writing the trailing raw makes this rule a reader of a run three other rules write, and its `always` cannot agree with `block-closing-brace-newline-before: never-multi-line` about that run under any spelling:

| the run in front of `}` | `block-opening-brace-newline-after: always` | `block-closing-brace-newline-before: never-multi-line` |
|---|---|---|
| nothing | reports | passes |
| a space | reports | passes |
| a line break | passes | reports, and takes the break out |

So a break written there would be taken straight back out, and the two rules would take it in turns for as long as `--fix` ran.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/block-opening-brace-newline-after": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

`--fix` clears the warning.

### What actually happened?

The warning stands, and the file comes back as it went in. It clears where `block-closing-brace-newline-before: always` is configured beside it, since that rule writes a break into the run — but on the run of `--fix` after the one that reported it, not the same one.

### Do you have a proposal to fix the bug?

The rule has to become a named reader of that run in the plugin's shared-run model, the way the two colon rules and the two semicolon rules are: who writes it, what the loser reports, and where the rule stands in the deferral order. `whitespaceAsked` and `neighbourSettings` are where that is asked today.

Found while working on issue #672.
